 import libraries

  import pickle 
 *  *pickle module implements binary protocols for serializing and de-serializing a* 
    *python object structure. “pickling” is the process whereby a python object hierarchy is*
    *converted into a byte stream*

In [1]:
import pandas as p
import numpy as np
import pickle
import emoji

 let's load some data!   

In [2]:
df=p.read_csv('condo_zhiv.csv')
df.head(3)

RegionID  SizeRank  RegionName RegionType StateName State     City  \
0     91940         0       77449        zip        TX    TX      NaN   
1     91982         1       77494        zip        TX    TX      NaN   
2     93144         2       79936        zip        TX    TX  El Paso   

                                  Metro        CountyName  2000-01-31  ...  \
0  Houston-The Woodlands-Sugar Land, TX     Harris County         NaN  ...   
1  Houston-The Woodlands-Sugar Land, TX  Fort Bend County         NaN  ...   
2                           El Paso, TX    El Paso County         NaN  ...   

   2021-11-30  2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  \
0    265501.0    269634.0    274911.0    275549.0    276192.0    274855.0   
1    282308.0    284093.0    286839.0    292454.0    299731.0    307873.0   
2    153779.0    156278.0    158503.0    161088.0    162301.0    164954.0   

   2022-05-31  2022-06-30  2022-07-31  2022-08-31  
0    281332.0    286178.0    292095.0    291221.0  
1    314752.0    318661.0    319232.0    318520.0  
2    168516.0    172793.0    175730.0    177317.0  

[3 rows x 281 columns]

 export metaData w pickles  🥒 

*  removing region data- iloc[:, 0:9]
*  check column (cols)
*     import os.mkdir()
   *  *mkdir is used for creating or making a new directory*
*     dump metaData
  *  'wb' for writing binary file. 
  *  'fp' file pointer

In [3]:
meta = df.iloc[:, 0:9]

In [4]:
meta.columns

Index(['RegionID', 'SizeRank', 'RegionName', 'RegionType', 'StateName',
       'State', 'City', 'Metro', 'CountyName'],
      dtype='object')

In [5]:
#import os
#os.mkdir('pickle')

In [5]:
with open('pickle/meta_condo.pkl', 'wb') as fp:
    pickle.dump(meta, fp)

 subsetting 
*  creating sac (sacramento) dataFrame
*  get list
*  remove col RegionName
*  drop cols (columns)


In [6]:
subset = df[df.CountyName == 'Sacramento County']

In [7]:
meta_cols = list(df.columns[0:9])

In [8]:
meta_cols.remove('RegionName')

In [9]:
subset = subset.drop(meta_cols, axis = 1)

 examine data 

In [10]:
subset.shape

(32, 273)

In [11]:
subset.head(3)

RegionName  2000-01-31  2000-02-29  2000-03-31  2000-04-30  2000-05-31  \
67        95823     47533.0     46659.0     46329.0     45116.0     44924.0   
91        95630    143509.0    143893.0    144090.0    144690.0    145729.0   
234       95624         NaN         NaN         NaN         NaN         NaN   

     2000-06-30  2000-07-31  2000-08-31  2000-09-30  ...  2021-11-30  \
67      44599.0     45375.0     46191.0     47818.0  ...    215039.0   
91     147086.0    149221.0    151225.0    153619.0  ...    453932.0   
234         NaN         NaN         NaN         NaN  ...    284628.0   

     2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
67     217432.0    221352.0    225397.0    229755.0    234188.0    238129.0   
91     457703.0    465728.0    474659.0    481736.0    488790.0    494852.0   
234    285970.0    291429.0    297831.0    303182.0    308305.0    311814.0   

     2022-06-30  2022-07-31  2022-08-31  
67     241279.0    242210.0    240197.0  
91     500375.0    498670.0    490328.0  
234    313683.0    314213.0    309188.0  

[3 rows x 273 columns]

 nulls? if so, how many?

In [12]:
subset.isna().sum().sum()

1460

 nulls? which columns?

In [13]:
subset.isna().any()

RegionName    False
2000-01-31     True
2000-02-29     True
2000-03-31     True
2000-04-30     True
              ...  
2022-04-30    False
2022-05-31    False
2022-06-30    False
2022-07-31    False
2022-08-31    False
Length: 273, dtype: bool

 nulls? which rows?

In [14]:
subset[subset.isna().any(axis=1)]

RegionName  2000-01-31  2000-02-29  2000-03-31  2000-04-30  2000-05-31  \
234        95624         NaN         NaN         NaN         NaN         NaN   
286        95758         NaN         NaN         NaN         NaN         NaN   
860        95757         NaN         NaN         NaN         NaN         NaN   
928        95843         NaN         NaN         NaN     96852.0     97672.0   
1640       95835         NaN         NaN         NaN         NaN         NaN   
1837       95838         NaN         NaN         NaN         NaN         NaN   
2107       95632         NaN         NaN         NaN         NaN         NaN   
2653       95662         NaN         NaN         NaN         NaN         NaN   
3025       95834         NaN         NaN         NaN         NaN         NaN   
4279       95827         NaN         NaN         NaN         NaN         NaN   
4403       95818         NaN         NaN         NaN         NaN         NaN   
5024       95816         NaN         NaN         NaN         NaN         NaN   
6673       95811         NaN         NaN         NaN         NaN         NaN   

      2000-06-30  2000-07-31  2000-08-31  2000-09-30  ...  2021-11-30  \
234          NaN         NaN         NaN         NaN  ...    284628.0   
286          NaN         NaN         NaN         NaN  ...    394701.0   
860          NaN         NaN         NaN         NaN  ...    451872.0   
928      98686.0    100600.0    101707.0    102502.0  ...    305832.0   
1640         NaN         NaN         NaN         NaN  ...    363576.0   
1837         NaN         NaN         NaN         NaN  ...    311445.0   
2107         NaN         NaN         NaN         NaN  ...    407443.0   
2653         NaN         NaN         NaN         NaN  ...    308664.0   
3025         NaN         NaN         NaN         NaN  ...    307779.0   
4279         NaN         NaN         NaN         NaN  ...    317459.0   
4403         NaN         NaN         NaN         NaN  ...    463354.0   
5024         NaN         NaN         NaN         NaN  ...    536414.0   
6673         NaN         NaN         NaN         NaN  ...    540962.0   

      2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
234     285970.0    291429.0    297831.0    303182.0    308305.0    311814.0   
286     397524.0    404793.0    412793.0    419122.0    424594.0    429781.0   
860     455711.0    466149.0    476901.0    485918.0    492478.0    498875.0   
928     306498.0    311587.0    316680.0    323612.0    328534.0    333966.0   
1640    366681.0    373711.0    380143.0    385861.0    390540.0    395687.0   
1837    309848.0    308464.0    309840.0    312850.0    318824.0    324233.0   
2107    408877.0    415954.0    422131.0    429813.0    433955.0    438908.0   
2653    310753.0    315553.0    318652.0    321247.0    323832.0    327818.0   
3025    312545.0    319847.0    326423.0    333116.0    338921.0    345926.0   
4279    319773.0    326285.0    334225.0    339918.0    343854.0    345690.0   
4403    462790.0    465456.0    471132.0    476800.0    483968.0    493613.0   
5024    533025.0    537985.0    543650.0    551966.0    557431.0    561804.0   
6673    539540.0    543833.0    550402.0    555656.0    561003.0    569409.0   

      2022-06-30  2022-07-31  2022-08-31  
234     313683.0    314213.0    309188.0  
286     432518.0    431971.0    423726.0  
860     503117.0    497856.0    485271.0  
928     337086.0    336153.0    329974.0  
1640    400454.0    399629.0    392704.0  
1837    328787.0    331151.0    328515.0  
2107    441963.0    441868.0    439333.0  
2653    330052.0    329144.0    323617.0  
3025    350952.0    351235.0    345803.0  
4279    348901.0    345235.0    336529.0  
4403    503018.0    506998.0    502957.0  
5024    558873.0    552153.0    541033.0  
6673    573695.0    571412.0    557707.0  

[13 rows x 273 columns]

lots of missing data up until jan 2008. Need to remove all prior too

In [15]:
subset.drop(subset.iloc[:,1:97], inplace=True, axis=1)

In [16]:
subset.head(3)

RegionName  2008-01-31  2008-02-29  2008-03-31  2008-04-30  2008-05-31  \
67        95823    133802.0    131579.0    128764.0    126656.0    122624.0   
91        95630    277974.0    273775.0    266485.0    260789.0    255940.0   
234       95624    186656.0    179289.0    172115.0    166903.0    161447.0   

     2008-06-30  2008-07-31  2008-08-31  2008-09-30  ...  2021-11-30  \
67     118191.0    111938.0    106887.0    100608.0  ...    215039.0   
91     250828.0    246367.0    241416.0    236327.0  ...    453932.0   
234    156799.0    153813.0    150956.0    145572.0  ...    284628.0   

     2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
67     217432.0    221352.0    225397.0    229755.0    234188.0    238129.0   
91     457703.0    465728.0    474659.0    481736.0    488790.0    494852.0   
234    285970.0    291429.0    297831.0    303182.0    308305.0    311814.0   

     2022-06-30  2022-07-31  2022-08-31  
67     241279.0    242210.0    240197.0  
91     500375.0    498670.0    490328.0  
234    313683.0    314213.0    309188.0  

[3 rows x 177 columns]

count of nulls remaining

In [17]:
subset.isna().sum().sum() 

413

which rows left w nulls

In [18]:
subset[subset.isna().any(axis=1)]

RegionName  2008-01-31  2008-02-29  2008-03-31  2008-04-30  2008-05-31  \
860        95757         NaN         NaN         NaN         NaN         NaN   
1837       95838         NaN    174371.0    167808.0    162408.0    157354.0   
4279       95827         NaN         NaN         NaN         NaN         NaN   
4403       95818         NaN         NaN         NaN         NaN         NaN   
5024       95816         NaN         NaN         NaN         NaN         NaN   

      2008-06-30  2008-07-31  2008-08-31  2008-09-30  ...  2021-11-30  \
860          NaN         NaN         NaN         NaN  ...    451872.0   
1837    152854.0    146471.0    138446.0    131049.0  ...    311445.0   
4279         NaN         NaN         NaN         NaN  ...    317459.0   
4403         NaN         NaN         NaN         NaN  ...    463354.0   
5024         NaN         NaN         NaN         NaN  ...    536414.0   

      2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
860     455711.0    466149.0    476901.0    485918.0    492478.0    498875.0   
1837    309848.0    308464.0    309840.0    312850.0    318824.0    324233.0   
4279    319773.0    326285.0    334225.0    339918.0    343854.0    345690.0   
4403    462790.0    465456.0    471132.0    476800.0    483968.0    493613.0   
5024    533025.0    537985.0    543650.0    551966.0    557431.0    561804.0   

      2022-06-30  2022-07-31  2022-08-31  
860     503117.0    497856.0    485271.0  
1837    328787.0    331151.0    328515.0  
4279    348901.0    345235.0    336529.0  
4403    503018.0    506998.0    502957.0  
5024    558873.0    552153.0    541033.0  

[5 rows x 177 columns]

of the 4 missing rows with data - 
1 row 95757 is an easy choice to eliminate. There aren't many if any condoss in that zip
95827, 18, and 16 are all right in the heart of sacramento and would be good to include, but no data until 2018. need to drop.

In [19]:
subset.drop([860,4279,4403,5024], axis=0, inplace=True)
subset

RegionName  2008-01-31  2008-02-29  2008-03-31  2008-04-30  2008-05-31  \
67         95823    133802.0    131579.0    128764.0    126656.0    122624.0   
91         95630    277974.0    273775.0    266485.0    260789.0    255940.0   
234        95624    186656.0    179289.0    172115.0    166903.0    161447.0   
286        95758    256688.0    247593.0    238262.0    230688.0    221088.0   
312        95608    209687.0    205596.0    199208.0    194521.0    190511.0   
489        95828    223392.0    219648.0    211567.0    203666.0    194934.0   
695        95670    256915.0    252203.0    244790.0    238195.0    233300.0   
928        95843    177058.0    173133.0    164889.0    159555.0    153945.0   
1049       95610    230984.0    227958.0    221193.0    216235.0    211726.0   
1118       95822    347985.0    344096.0    333530.0    326926.0    319430.0   
1305       95831    240674.0    236616.0    230652.0    226992.0    224934.0   
1343       95628    220956.0    217416.0    211028.0    205522.0    200808.0   
1540       95621    200965.0    196671.0    188699.0    182475.0    177167.0   
1640       95835    242168.0    237020.0    223694.0    212541.0    201429.0   
1694       95833    214194.0    211752.0    206083.0    201619.0    197883.0   
1716       95826    172039.0    168596.0    162599.0    158069.0    154374.0   
1812       95821    181790.0    177862.0    172671.0    169300.0    166536.0   
1837       95838         NaN    174371.0    167808.0    162408.0    157354.0   
2078       95825    235793.0    230673.0    223225.0    217846.0    213802.0   
2107       95632    239334.0    236015.0    232652.0    225362.0    217830.0   
2323       95842    176263.0    171678.0    163985.0    158003.0    152396.0   
2653       95662    187326.0    186515.0    182649.0    177305.0    169765.0   
3025       95834    221575.0    216820.0    209373.0    201677.0    194127.0   
4066       95864    398713.0    391191.0    380798.0    372232.0    366791.0   
4561       95841    150816.0    148098.0    142573.0    138253.0    135439.0   
6202       95814    434479.0    426243.0    408132.0    394551.0    386681.0   
6673       95811    363798.0    358181.0    348245.0    342489.0    338753.0   
6892       95683    364938.0    354239.0    339433.0    330048.0    322503.0   

      2008-06-30  2008-07-31  2008-08-31  2008-09-30  ...  2021-11-30  \
67      118191.0    111938.0    106887.0    100608.0  ...    215039.0   
91      250828.0    246367.0    241416.0    236327.0  ...    453932.0   
234     156799.0    153813.0    150956.0    145572.0  ...    284628.0   
286     213029.0    206886.0    203193.0    196815.0  ...    394701.0   
312     185951.0    181749.0    176014.0    170118.0  ...    319327.0   
489     187536.0    181223.0    176028.0    170296.0  ...    373435.0   
695     227772.0    221730.0    213214.0    203360.0  ...    354777.0   
928     149730.0    143503.0    137794.0    133346.0  ...    305832.0   
1049    208245.0    205370.0    201076.0    194728.0  ...    350843.0   
1118    315347.0    308368.0    303593.0    295987.0  ...    432900.0   
1305    223488.0    219947.0    215087.0    208538.0  ...    356987.0   
1343    196409.0    194047.0    190651.0    186775.0  ...    338774.0   
1540    172234.0    165937.0    159372.0    153158.0  ...    310069.0   
1640    195444.0    189971.0    186034.0    180746.0  ...    363576.0   
1694    195673.0    192640.0    187505.0    178993.0  ...    364852.0   
1716    150334.0    144703.0    140010.0    135882.0  ...    271044.0   
1812    162741.0    158304.0    152328.0    146457.0  ...    269220.0   
1837    152854.0    146471.0    138446.0    131049.0  ...    311445.0   
2078    208940.0    204797.0    199829.0    194579.0  ...    322934.0   
2107    209580.0    203927.0    199147.0    194897.0  ...    407443.0   
2323    149099.0    144983.0    138391.0    128682.0  ...    205183.0   
2653    166113.0    166414.0    165333.0    162207.0  ...    308664.0   
3025  

rows w nans

In [20]:
subset[subset.isna().any(axis=1)]

RegionName  2008-01-31  2008-02-29  2008-03-31  2008-04-30  2008-05-31  \
1837       95838         NaN    174371.0    167808.0    162408.0    157354.0   

      2008-06-30  2008-07-31  2008-08-31  2008-09-30  ...  2021-11-30  \
1837    152854.0    146471.0    138446.0    131049.0  ...    311445.0   

      2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
1837    309848.0    308464.0    309840.0    312850.0    318824.0    324233.0   

      2022-06-30  2022-07-31  2022-08-31  
1837    328787.0    331151.0    328515.0  

[1 rows x 177 columns]

count of nans

In [21]:
subset.isna().sum().sum() 

1

1 nan. lets bfill it

In [22]:
subset.bfill(axis=0, inplace=True, limit=2)

In [23]:
subset.isna().sum().sum()

0

 quick last check of df

In [24]:
subset.head()

RegionName  2008-01-31  2008-02-29  2008-03-31  2008-04-30  2008-05-31  \
67        95823    133802.0    131579.0    128764.0    126656.0    122624.0   
91        95630    277974.0    273775.0    266485.0    260789.0    255940.0   
234       95624    186656.0    179289.0    172115.0    166903.0    161447.0   
286       95758    256688.0    247593.0    238262.0    230688.0    221088.0   
312       95608    209687.0    205596.0    199208.0    194521.0    190511.0   

     2008-06-30  2008-07-31  2008-08-31  2008-09-30  ...  2021-11-30  \
67     118191.0    111938.0    106887.0    100608.0  ...    215039.0   
91     250828.0    246367.0    241416.0    236327.0  ...    453932.0   
234    156799.0    153813.0    150956.0    145572.0  ...    284628.0   
286    213029.0    206886.0    203193.0    196815.0  ...    394701.0   
312    185951.0    181749.0    176014.0    170118.0  ...    319327.0   

     2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
67     217432.0    221352.0    225397.0    229755.0    234188.0    238129.0   
91     457703.0    465728.0    474659.0    481736.0    488790.0    494852.0   
234    285970.0    291429.0    297831.0    303182.0    308305.0    311814.0   
286    397524.0    404793.0    412793.0    419122.0    424594.0    429781.0   
312    321546.0    326917.0    333961.0    340607.0    347925.0    353805.0   

     2022-06-30  2022-07-31  2022-08-31  
67     241279.0    242210.0    240197.0  
91     500375.0    498670.0    490328.0  
234    313683.0    314213.0    309188.0  
286    432518.0    431971.0    423726.0  
312    358226.0    357076.0    350995.0  

[5 rows x 177 columns]

 calculate percent increase from yr to yr in sac
*  create df and define func (function)
*  create copy 
   *  copy() method creates a copy of an existing list

In [25]:
def calculate_percent_increase(x1, x2):
    return ((x2-x1)/x1)*100

In [26]:
subset_percent = subset.copy()
for i in range(subset_percent.shape[1]-1,1, -1):
    prior = subset_percent.iloc[:, i-1]
    current = subset_percent.iloc[:, i] 
    subset_percent.iloc[:, i] = calculate_percent_increase(prior, current)

In [27]:
subset_percent.head(3)

RegionName  2008-01-31  2008-02-29  2008-03-31  2008-04-30  2008-05-31  \
67        95823    133802.0   -1.661410   -2.139399   -1.637104   -3.183426   
91        95630    277974.0   -1.510573   -2.662771   -2.137456   -1.859358   
234       95624    186656.0   -3.946833   -4.001361   -3.028208   -3.268965   

     2008-06-30  2008-07-31  2008-08-31  2008-09-30  ...  2021-11-30  \
67    -3.615116   -5.290589   -4.512319   -5.874428  ...    0.513695   
91    -1.997343   -1.778510   -2.009604   -2.107980  ...    0.747731   
234   -2.878963   -1.904349   -1.857450   -3.566602  ...    0.574199   

     2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
67     1.112821    1.802863    1.827406    1.933477    1.929447    1.682836   
91     0.830741    1.753320    1.917643    1.490965    1.464287    1.240205   
234    0.471493    1.908941    2.196761    1.796656    1.689744    1.138159   

     2022-06-30  2022-07-31  2022-08-31  
67     1.322812    0.385860   -0.831097  
91     1.116091   -0.340744   -1.672850  
234    0.599396    0.168960   -1.599234  

[3 rows x 177 columns]

In [28]:
subset_percent.drop('2008-01-31' ,axis=1, inplace=True)

In [29]:
subset_percent.head(3)

RegionName  2008-02-29  2008-03-31  2008-04-30  2008-05-31  2008-06-30  \
67        95823   -1.661410   -2.139399   -1.637104   -3.183426   -3.615116   
91        95630   -1.510573   -2.662771   -2.137456   -1.859358   -1.997343   
234       95624   -3.946833   -4.001361   -3.028208   -3.268965   -2.878963   

     2008-07-31  2008-08-31  2008-09-30  2008-10-31  ...  2021-11-30  \
67    -5.290589   -4.512319   -5.874428   -3.802878  ...    0.513695   
91    -1.778510   -2.009604   -2.107980   -0.807779  ...    0.747731   
234   -1.904349   -1.857450   -3.566602   -4.008326  ...    0.574199   

     2021-12-31  2022-01-31  2022-02-28  2022-03-31  2022-04-30  2022-05-31  \
67     1.112821    1.802863    1.827406    1.933477    1.929447    1.682836   
91     0.830741    1.753320    1.917643    1.490965    1.464287    1.240205   
234    0.471493    1.908941    2.196761    1.796656    1.689744    1.138159   

     2022-06-30  2022-07-31  2022-08-31  
67     1.322812    0.385860   -0.831097  
91     1.116091   -0.340744   -1.672850  
234    0.599396    0.168960   -1.599234  

[3 rows x 176 columns]

 check df  percent increase

 transpose data
*  changing to wide format

In [30]:
subset_percent = subset_percent.set_index('RegionName').transpose()

In [31]:
subset = subset.set_index('RegionName').transpose()

In [32]:
subset_percent.head(3)

RegionName     95823     95630     95624     95758     95608     95828  \
2008-02-29 -1.661410 -1.510573 -3.946833 -3.543212 -1.951003 -1.675978   
2008-03-31 -2.139399 -2.662771 -4.001361 -3.768685 -3.107064 -3.679068   
2008-04-30 -1.637104 -2.137456 -3.028208 -3.178854 -2.352817 -3.734514   

RegionName     95670     95843     95610     95822  ...     95825     95632  \
2008-02-29 -1.834070 -2.216788 -1.310047 -1.117577  ... -2.171396 -1.386765   
2008-03-31 -2.939299 -4.761657 -2.967652 -3.070655  ... -3.228813 -1.424909   
2008-04-30 -2.694146 -3.234903 -2.241481 -1.980032  ... -2.409676 -3.133435   

RegionName     95842     95662     95834     95864     95841     95814  \
2008-02-29 -2.601227 -0.432935 -2.146000 -1.886570 -1.802196 -1.895604   
2008-03-31 -4.481063 -2.072756 -3.434646 -2.656758 -3.730638 -4.248985   
2008-04-30 -3.647895 -2.925830 -3.675737 -2.249487 -3.030027 -3.327600   

RegionName     95811     95683  
2008-02-29 -1.543989 -2.931731  
2008-03-31 -2.774016 -4.179664  
2008-04-30 -1.652859 -2.764905  

[3 rows x 28 columns]

In [33]:
subset.head(3)

RegionName     95823     95630     95624     95758     95608     95828  \
2008-01-31  133802.0  277974.0  186656.0  256688.0  209687.0  223392.0   
2008-02-29  131579.0  273775.0  179289.0  247593.0  205596.0  219648.0   
2008-03-31  128764.0  266485.0  172115.0  238262.0  199208.0  211567.0   

RegionName     95670     95843     95610     95822  ...     95825     95632  \
2008-01-31  256915.0  177058.0  230984.0  347985.0  ...  235793.0  239334.0   
2008-02-29  252203.0  173133.0  227958.0  344096.0  ...  230673.0  236015.0   
2008-03-31  244790.0  164889.0  221193.0  333530.0  ...  223225.0  232652.0   

RegionName     95842     95662     95834     95864     95841     95814  \
2008-01-31  176263.0  187326.0  221575.0  398713.0  150816.0  434479.0   
2008-02-29  171678.0  186515.0  216820.0  391191.0  148098.0  426243.0   
2008-03-31  163985.0  182649.0  209373.0  380798.0  142573.0  408132.0   

RegionName     95811     95683  
2008-01-31  363798.0  364938.0  
2008-02-29  358181.0  354239.0  
2008-03-31  348245.0  339433.0  

[3 rows x 28 columns]

 Let's convert the index to datetime!

*  strftime() function is used to convert to Index using specified date_format.
*   columns.name=none- names the index, by default it is None or blank, the columns are of type IndexArray

In [34]:
subset_percent.index = p.DatetimeIndex(subset_percent.index)
subset_percent.index = subset_percent.index.strftime('%m/%Y')
subset_percent.columns.name = None

In [35]:
subset.index = p.DatetimeIndex(subset.index)
subset.index = subset.index.strftime('%m/%Y')
subset.columns.name = None

 exporting 🥒


In [36]:
with open('pickle/timeseries_condo_subset_percent.pkl', 'wb') as fp:
    pickle.dump(subset_percent, fp)

In [37]:
with open('pickle/timeseries_condo_subset.pkl', 'wb') as fp:
    pickle.dump(subset, fp)